### PR TITLE
Dala 1429 update datapoint upload forms and Dala 1409 useful redirect to MyDatasets page

### PR DESCRIPTION
**Pull request by janherre**
# Pull Request \<Title>
## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
- [x] The PR actually implements what is described in the JIRA-Issue
- [x] At least one E2E Test exists testing the new feature
- [x] Documentation is updated as required
- [x] The automated deployment is updated if required
- [x] The new version is deployed to the dev server "dev1" using this branch
  - [x] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [x] The new feature is manually used/tested/observed on dev server
  - [x] All implemented Social Logins have been tested manually in the UI
- [x] If any work on the UI is to be merged, those changes were also documented in the Figma

### CODE REVIEW

File: dataland-frontend/src/components/resources/uploadDataset/datasetCreationRedirect.ts
Comment (fsoffel) [inline dataland-frontend/src/components/resources/uploadDataset/datasetCreationRedirect.ts:1]: Do we do camel case naming for typescript files?

File: dataland-frontend/src/components/resources/uploadDataset/datasetCreationRedirect.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import { Router } from "vue-router";
+
+/**
+ * function that redirects from the current page to the myDatasets page.
+ * It is called after successfully submitting a newly created dataset
+ *
+ * @param [router] the router of the Vue Component that calls the function
+ */
+export function redirectToMyDatasets(router: Router): void {
+  setTimeout(() => {
```
Comment (fsoffel) [inline dataland-frontend/src/components/resources/uploadDataset/datasetCreationRedirect.ts:11]: Why do we need the delay?

File: dataland-frontend/src/components/forms/parts/kpiSelection/DataPointForm.vue [hunk 2]
```diff
@@ -147,6 +151,10 @@ export default defineComponent({
       type: String,
       default: "percent",
     },
+    reportsName: {
```
Comment (fsoffel) [inline dataland-frontend/src/components/forms/parts/kpiSelection/DataPointForm.vue:154]: This seems unused

File: dataland-frontend/src/components/pages/UploadEuTaxonomyDataForNonFinancials.vue [hunk 3]
```diff
@@ -46,5 +51,11 @@ export default {
       type: String,
     },
   },
-};
+  methods: {
+    /**
+     * Executes a router push to the myDatasets page
+     */
+    redirectToMyDatasets,
```
Comment (fsoffel) [inline dataland-frontend/src/components/pages/UploadEuTaxonomyDataForNonFinancials.vue:58]: You can also put this in the data block. Then you don't need the additional documentation. But good to know that it also works in the methods block.

File: dataland-frontend/src/components/forms/parts/kpiSelection/DataPointForm.vue [hunk 4]
```diff
@@ -84,7 +85,6 @@
     <FormKit
       type="textarea"
       name="comment"
-      rows="10"
```
Comment (fsoffel) [inline dataland-frontend/src/components/forms/parts/kpiSelection/DataPointForm.vue:87]: This is aligned with UX/UI people?

File: dataland-frontend/src/components/forms/CreateEUTaxonomyForFinancials.vue [hunk 5]
```diff
@@ -80,7 +80,7 @@
                         >
                           <span data-test="uploaded-files-title" class="font-semibold flex-1">{{ file.name }}</span>
                           <div data-test="uploaded-files-size" class="mx-2 text-black-alpha-50">
-                            {{ formatSize(file.size) }}
+                            {{ formatSize(Number(file.size)) }}
```
Comment (fsoffel) [inline dataland-frontend/src/components/forms/CreateEUTaxonomyForFinancials.vue:83]: What is the type of file.size?

File: dataland-frontend/src/components/forms/CreateEUTaxonomyForFinancials.vue [hunk 6]
```diff
@@ -376,40 +376,38 @@
...
                         <FormKit :name="kpiType" type="group">
                           <div class="form-field">
-                            <UploadFormHeader
-                              :name="euTaxonomyKpiNameMappings[kpiType] ?? ''"
-                              :explanation="euTaxonomyKpiInfoMappings[kpiType] ?? ''"
-                            />
-                            <KPIfieldSet
+                            <DataPointHeader :name="euTaxonomyKpiNameMappings[kpiType]" />
+                            <DataPointForm
+                              :name="kpiType"
```
Comment (fsoffel) [inline dataland-frontend/src/components/forms/CreateEUTaxonomyForFinancials.vue:410]: Why not join these two components, they only occur in together.

File: dataland-frontend/src/components/forms/CreateEUTaxonomyForNonFinancials.vue [hunk 7]
```diff
@@ -24,7 +24,7 @@
             />
             <div class="uploadFormSection grid">
               <div class="col-3 p-3 topicLabel">
-                <h4 id="uploadReports" class="anchor title">Reporting Period</h4>
+                <h4 id="reportingPeriod" class="anchor title">Reporting Period</h4>
```
Comment (fsoffel) [inline dataland-frontend/src/components/forms/CreateEUTaxonomyForNonFinancials.vue:27]: This also occurs on the financials page.

File: dataland-frontend/src/components/forms/CreateEUTaxonomyForNonFinancials.vue [hunk 8]
```diff
@@ -42,14 +42,11 @@
                   />
                 </div>
 
-                <FormKit
```
Comment (fsoffel) [inline dataland-frontend/src/components/forms/CreateEUTaxonomyForNonFinancials.vue:45]: removed due to reduncancy?

File: dataland-frontend/tests/e2e/utils/EuTaxonomyNonFinancialsUpload.ts [hunk 9]
```diff
@@ -50,14 +50,14 @@ export function uploadEuTaxonomyDataForNonFinancialsViaForm(
...
   cy.get('input[name="activityLevelReporting"][value="Yes"]').check();
-  cy.get('[data-test="dataPointToggleTitle"]').should("not.exist");
   cy.get('input[name="numberOfEmployees"]').type("333");
   cy.get('input[name="reportingObligation"][value="Yes"]').check();
 
   cy.get('[data-test="assuranceSection"] select[name="assurance"]').select(1);
   cy.get('[data-test="assuranceSection"] input[name="provider"]').type("Assurance Provider");
   cy.get('[data-test="assuranceSection"] select[name="report"]').select(1);
 
+  cy.get('[data-test="dataPointToggleTitle"]').should("exist");
```
Comment (fsoffel) [inline dataland-frontend/tests/e2e/utils/EuTaxonomyNonFinancialsUpload.ts:60]: Do we test the function of the toggle somewhere?

File: dataland-frontend/src/components/forms/CreateEUTaxonomyForNonFinancials.vue [hunk 10]
```diff
@@ -344,28 +341,25 @@
...
                     <FormKit name="capex" type="group">
                       <div
                         v-for="detailCashFlowType of euTaxonomyKPIsModel.euTaxonomyDetailsPerCashFlowType"
                         :key="detailCashFlowType"
                         :data-test="detailCashFlowType"
-                        class="form-field"
+                        class="formFields"
                       >
-                        <FormKit :name="detailCashFlowType" type="group">
+                        <FormKit
```
Comment (fsoffel) [inline dataland-frontend/src/components/forms/CreateEUTaxonomyForNonFinancials.vue:352]: Why do you do this redesign? I can not find the corresponding figma.

File: dataland-frontend/src/components/resources/uploadDataset/datasetCreationRedirect.ts
Comment (janherre) [reply dataland-frontend/src/components/resources/uploadDataset/datasetCreationRedirect.ts:1]: Yes, but first letter should be capitalized, I will change it

File: dataland-frontend/src/components/forms/parts/kpiSelection/DataPointForm.vue (hunk 2 above)
Comment (janherre) [reply dataland-frontend/src/components/forms/parts/kpiSelection/DataPointForm.vue:154]: Removed, could be introduced by the upload documents story, there were some changes merged with cherrypicking

File: dataland-frontend/src/components/forms/parts/kpiSelection/euTaxonomyKPIsModel.ts
[the remark this answers is not in the document]
Comment (janherre) [reply dataland-frontend/src/components/forms/parts/kpiSelection/euTaxonomyKPIsModel.ts:1]: Not introduced in this story, as far as I can see all ts files are named like this, First letter should be capitalized probably.

File: dataland-frontend/src/components/pages/UploadEuTaxonomyDataForNonFinancials.vue (hunk 3 above)
Comment (janherre) [reply dataland-frontend/src/components/pages/UploadEuTaxonomyDataForNonFinancials.vue:58]: Changed. Intellij automatically puts it in the methods block on import.

Comment (fsoffel) [reply dataland-frontend/src/components/pages/UploadEuTaxonomyDataForNonFinancials.vue]: Aah. I mean its also fine. But I was just curious

File: dataland-frontend/src/components/forms/CreateEUTaxonomyForFinancials.vue (hunk 5 above)
Comment (fsoffel) [reply dataland-frontend/src/components/forms/CreateEUTaxonomyForFinancials.vue]: https://github.com/d-fine/Dataland/pull/211#discussion_r1168355742

Comment (fsoffel) [reply dataland-frontend/src/components/forms/CreateEUTaxonomyForFinancials.vue]: I mean, isn't it number by default?

File: dataland-frontend/src/components/forms/CreateEUTaxonomyForFinancials.vue
Comment (fsoffel) [inline dataland-frontend/src/components/forms/CreateEUTaxonomyForFinancials.vue:1]: Line 25 https://github.com/d-fine/Dataland/pull/211#discussion_r1168309681

File: dataland-frontend/src/components/forms/CreateEUTaxonomyForNonFinancials.vue [hunk 11]
```diff
@@ -87,7 +84,7 @@
                         >
                           <span data-test="uploaded-files-title" class="font-semibold flex-1">{{ file.name }}</span>
                           <div data-test="uploaded-files-size" class="mx-2 text-black-alpha-50">
-                            {{ formatSize(file.size) }}
+                            {{ formatBytesUserFriendly(file.size, 3) }}
```
[the remark this answers is not in the document]
Comment (fsoffel) [reply dataland-frontend/src/components/forms/CreateEUTaxonomyForNonFinancials.vue]: Nicht im remote repository :)

File: dataland-frontend/tests/e2e/utils/EuTaxonomyNonFinancialsUpload.ts (hunk 9 above)
Comment (fsoffel) [reply dataland-frontend/tests/e2e/utils/EuTaxonomyNonFinancialsUpload.ts:60]: Probably.

File: dataland-frontend/src/components/forms/CreateEUTaxonomyForNonFinancials.vue (hunk 11 above)
[the remark this answers is not in the document]
Comment (janherre) [reply dataland-frontend/src/components/forms/CreateEUTaxonomyForNonFinancials.vue]: Doch, in folgendem commit 98dd436

File: dataland-frontend/src/components/forms/CreateEUTaxonomyForFinancials.vue
Comment (janherre) [reply dataland-frontend/src/components/forms/CreateEUTaxonomyForFinancials.vue:1]: Was meinst du hier?

File: dataland-frontend/src/components/forms/CreateEUTaxonomyForNonFinancials.vue (hunk 11 above)
[the remark this answers is not in the document]
Comment (fsoffel) [reply dataland-frontend/src/components/forms/CreateEUTaxonomyForNonFinancials.vue]: Lol... Die aufgerufene Funktion ist doch jetzt quasi dead. Also weg damit.

**Comment by fsoffel** [comment]
uploadReports ist als id angegeben beim Titel "Reporting Periods". Ist ein bug, ids dürfen nicht zweimal vergeben werden.

**Pull request merged**